tools/build_defs/repo/utils.bzl

```diff
@@ -75,26 +87,45 @@ def patch(ctx):
         function.
     """
     bash_exe = ctx.os.environ["BAZEL_SH"] if "BAZEL_SH" in ctx.os.environ else "bash"
+    powershell_exe = ctx.os.environ["BAZEL_POWERSHELL"] if "BAZEL_POWERSHELL" in ctx.os.environ else "powershell.exe"
     if len(ctx.attr.patches) > 0 or len(ctx.attr.patch_cmds) > 0:
         ctx.report_progress("Patching repository")
-    for patchfile in ctx.attr.patches:
-        command = "{patchtool} {patch_args} < {patchfile}".format(
-            patchtool = ctx.attr.patch_tool,
-            patchfile = ctx.path(patchfile),
-            patch_args = " ".join([
-                "'%s'" % arg
-                for arg in ctx.attr.patch_args
-            ]),
-        )
-        st = ctx.execute([bash_exe, "-c", command])
-        if st.return_code:
-            fail("Error applying patch %s:\n%s%s" %
-                 (str(patchfile), st.stderr, st.stdout))
-    for cmd in ctx.attr.patch_cmds:
-        st = ctx.execute([bash_exe, "-c", cmd])
-        if st.return_code:
-            fail("Error applying patch command %s:\n%s%s" %
-                 (cmd, st.stdout, st.stderr))
+
+    if _use_native_patch(ctx):
+        for arg in ctx.attr.patch_args:
+            strip = int(arg[2:])
```

Comment:
Please assert the argument starts with "-p"
Also, what if there is patch -p1 -p2 -p3 - then the last option wins.
Maybe better allow only one option to be specified.

Reply:
`_use_native_patch` already ensures all arguments starts with `-p`.
The patch command line tool also uses the value of the last specified `-p` option. Maybe we should keep the same behavior?

Reply:
Well, ok, the assertion part looks solid,

but with applying the last of the -p options, could we add a comment about that and explicitly get the last of the patch arguments? Otherwise, it is not obvious for the code reader.

Reply:
Good point, I documented the behavior of multiple `-p` options.

Reply:
It is still an iteratino here, Yun. Why do we need to iterate?))))

Reply:
Ah, what was I thinking.. Fixed now, thanks!

Reply:
Oh, I remember, I was thinking about parsing other options like `-e` as well. But later, decided to keep it simple.